R/get_file_metadata.R

```diff
@@ -0,0 +1,98 @@
+#' Get Metadata for a vector of filepaths
+#'
+#' This function takes a vector of filepaths and returns a list of file
+#' details, suitable for inclusion in manifest export.
+#'
+#' @param filepaths vector of filepaths
+#'
+#' @return nested list of file details, length the same as the input vector.
+get_file_metadata <- function(filepaths) {
+  logger::log_trace("Getting metadata for files.")
+
+  file_metadata <- lapply(filepaths, get_single_file_metadata)
+
+  return(file_metadata)
+}
+
+#' Get Metadata for a file
+#'
+#' This function takes a single filepaths and returns a list of file
+#' details, suitable for inclusion in manifest export.
+#'
+#' @param filepath vector of filepaths
+#'
+#' @return list of file details
+get_single_file_metadata <- function(filepath) {
+  if (length(filepath) > 1L) {
+    logger::log_error("get_single_file_metadata only accepts single files.")
+    stop("Only one file path can be passed to get_single_file_metadata.")
+  }
+  if (!file.exists(filepath)) {
+    logger::log_error("File does not exist: \"{filepath}\".")
+    stop("File does not exist.")
+  }
+
+  logger::log_trace("Getting metadata for file: \"{filepath}\".")
+
+  file_name <- basename(filepath)
+  file_extension <- tools::file_ext(filepath)
+  file_path <- filepath
+  file_size <- as.integer(file.info(filepath)[["size"]])
+  file_last_modified <- format(
+    as.POSIXlt(file.info(filepath)[["mtime"]], tz = "UTC"),
+    "%Y-%m-%dT%H:%M:%S+00:00"
+  )
+  file_md5 <- digest::digest(filepath, algo = "md5", file = TRUE)
+
+  file_metadata <- list(
+    file_name = file_name,
+    file_extension = file_extension,
+    file_path = file_path,
+    file_size = file_size,
+    file_last_modified = file_last_modified,
+    file_md5 = file_md5
+  )
+
+  logger::log_trace("Getting summary information for file: \"{filepath}\".")
+  if (tolower(tools::file_ext(filepath)) == "rds") {
+    contents <- readRDS(filepath)
+  } else if (tolower(tools::file_ext(filepath)) == "csv") {
+    contents <- utils::read.csv(filepath)
```

Comment:
might need to worry about encodings here

Reply:
Possibly, but I'd prefer to deal with that if we discover the issue, and let the R default handle it for now, since we're not dealing with the contents of the file beyond colnames and rowcounts, which should aonly cause issues if there's an encoding issue in the header.